internal/parsers/collapsed/collapsed_parser.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collapsed
+
+import (
+  "bufio"
+  "fmt"
+  "io"
+  "strconv"
+  "strings"
+
+  "github.com/google/instrumentsToPprof/internal"
+)
+
+func MakeCollapsedParser(file io.Reader) (d CollapsedParser, err error) {
+  d = CollapsedParser{
+    lines: []string{},
+  }
+  scanner := bufio.NewScanner(file)
+  for scanner.Scan() {
+    d.lines = append(d.lines, scanner.Text())
+  }
+  if err := scanner.Err(); err != nil {
+    return d, err
+  }
+  return d, nil
+}
+
+// This parser supports the "collapsed stack" file format where each line of 
+// the file represents a full stack sample. This is a format commonly used
+// to generate flamegraphs.
+type CollapsedParser struct {
+  lines []string
+}
+
+func (d CollapsedParser) ParseProfile() (p *internal.TimeProfile, err error) {
+  p = &internal.TimeProfile{}
+
+  // Collapsed format merges all processes so just create a dummy one.
```

Comment:
If collapsed merges everything, why not just use a pprof profile directly? This doesn't work in the normal abstraction for the other types but I would think it would be much cleaner.

Reply:
The reason why I did it like that in this instance is that while it doesn't have process and thread information right now it's possible that I could add it eventually.

I actually personally thought it was cleaner to use the TimeProfile to be consistent with the other parsers.

If at all possible I'd really rather not have to depart too much from the implementation you provided for lack of time.

Would asserting that excludeProcessInStack and excludeThreadsInStack are specified when using the collapsed format sound good?

I would not make the code much different but it would avoid funky looking profiles.